datafusion/sqllogictest/test_files/arrow_typeof.slt

```diff
@@ -95,9 +95,12 @@ SELECT arrow_cast('1', 'Int16')
 query error
 SELECT arrow_cast('1')
 
-query error DataFusion error: Execution error: arrow_cast requires its second argument to be a non\-empty constant string
+query error Expect TypeSignatureClass::Native\(LogicalType\(Native\(String\), String\)\) but received NativeType::Int64, DataType: Int64
```

Comment:
Probably should start looking in https://github.com/apache/datafusion/issues/19004 properly to try pretty up these new error messages 🤔

Reply:
These are really ugly I agree it would be nice if it was neater. But the current status quo is not good so this is no better or worse imo.